Client/src/Unified/Factory/ServiceFactory.cs

```diff
@@ -37,17 +38,19 @@ public class ServiceFactory
   /// <param name="props">Properties for the service containing IConfiguration and TaskOptions</param>
   /// <param name="loggerFactory">Logger factory to create loggers for service</param>
   /// <returns>returns the new instantiated service</returns>
-  public static ISubmitterService CreateService(Properties      props,
-                                                ILoggerFactory? loggerFactory = null)
-    => new Service(props,
-                   loggerFactory);
+  [PublicAPI]
+  public static Service CreateService(Properties      props,
```

Comment:
Why not return the interface ?

Reply:
It's a factory for Unified, so it should return the Service for Unified

Reply:
It's a returning type so it's less important however, I prefer that we force the users to only manipulate APIs instance. This way, we can more easily provide invisible code change.

Reply:
If upcasting wasn't explicit that wouldn't be an issue. I don't mind returning the interface as long as we state the change explicitely in the changelog (because it changes the front facing client view)